assets/app/view/game_card.rb

```diff
@@ -42,6 +45,24 @@ def acting?(player)
       acting.include?(player[:id])
     end
 
+    def render_private_join
+      return h(:div, []) unless @is_joining == @gdata['id']
+
+      password_handler = ->(event) { store(:password, Native(event)['target']['value']) }
+      join_handler = lambda do
+        store(:is_joining, nil)
+        join_game(@gdata, @password)
+      end
+      input_params = {
+        attrs: { type: 'text', placeholder: 'Password' },
+        on: { input: password_handler },
```

Comment:
every time you do an input this gets called and does a full refresh of the screen, maybe instead join_game should just pull the native value of this